pkg/metadata/metadata_store.go

```diff
@@ -0,0 +1,41 @@
+package metadata
+
+import (
+	"sync"
+)
+
+// Store is an interface to perform operations on metadata store.
+type Store interface {
+	MetadataGet() (map[string]string, error)
+	MetadataSet(key, value string) error
+}
+
+// DefaultStore is a default implementation of Store.
+type DefaultStore struct {
+	metadata sync.Map
+}
+
+// NewDefaultStore build a default metadata store.
+func NewDefaultStore() DefaultStore {
+	return DefaultStore{}
+}
+
+// MetadataGet get metadata of sidecar.
+func (m *DefaultStore) MetadataGet() (map[string]string, error) {
```

Comment:
Always nil error? Could we remove it?

Reply:
If the follow-up is no longer in-memory storage, it may no longer be a nil error.

Reply:
Is there any plan metadata stored in something else rather than in-memory storage?